src/main/java/org/broadinstitute/consent/http/service/TDRService.java

```diff
@@ -33,4 +38,18 @@ public ApprovedUsers getApprovedUsersForDataset(Dataset dataset) {
 
         return new ApprovedUsers(approvedUsers);
     }
+
+    public List<Integer> getDatasetIdsByIdentifier(List<String> identifiers) {
+        List<Integer> datasetIds = identifiers
+                .stream()
+                .filter(identifier -> !identifier.isBlank())
+                .map(identifier -> datasetDAO.findDatasetByAlias(Dataset.parseIdentifierToAlias(identifier)))
+                // technically, it is possible to have two dataset identifiers which
+                // have the same alias but are not the same: e.g., DUOS-5 and DUOS-00005
+                .filter(d -> !identifiers.contains(d.getDatasetIdentifier()))
+                .map(d -> d.getDataSetId())
```

Comment:
```suggestion
                .map(Dataset::getDataSetId)
```

Reply:
Refactored all new map calls to this syntax.